Userland/Applications/Piano/WaveWidget.cpp

```diff
@@ -9,24 +9,40 @@
 #include "WaveWidget.h"
 #include "TrackManager.h"
 #include <AK/NumericLimits.h>
+#include <AK/StdLibExtras.h>
 #include <LibGUI/Painter.h>
 
 WaveWidget::WaveWidget(TrackManager& track_manager)
     : m_track_manager(track_manager)
 {
 }
 
-int WaveWidget::sample_to_y(int sample) const
+int WaveWidget::sample_to_y(float sample, float sample_max) const
 {
-    // Sample scaling that looks good, experimentally determined.
-    constexpr double nice_scale_factor = 1.0;
-    constexpr double sample_max = NumericLimits<i16>::max();
-    double percentage = sample / sample_max * nice_scale_factor;
+    if (sample_max < 1.0f)
+        sample_max = 1.0f;
+    sample_max *= 1.2f;
+    double percentage = static_cast<double>(sample) / static_cast<double>(sample_max);
     double portion_of_half_height = percentage * ((frame_inner_rect().height() - 1) / 2.0);
     double y = (frame_inner_rect().height() / 2.0) + portion_of_half_height;
     return y;
 }
 
+static float absf(float v)
+{
+    return v < 0 ? -v : v;
+}
```

Comment:
This should not live here; if it's missing from AK/Math or LibC/math please add it there.